app/views/hours_reports/index.html.haml

```diff
@@ -0,0 +1,55 @@
+- set_headline  "Listing Users' Hours"
+= javascript_include_tag "/flot/jquery.flot.js"
+= javascript_include_tag "/flot/jquery.flot.symbol.js"
+= javascript_include_tag "/flot/jquery.flot.axislabels.js"
+
+.button-wrap
+  %div.button.radius#all-user-hours-btn
+    All
+  %div.button.radius#total-user-hours-btn
+    Total
+  %div.button.radius#billable-user-hours-btn
+    Billable
+  %div.button.radius#unbillable-user-hours-btn
+    Unbillable
+
+= form_tag hours_reports_path do
+  %div.input-container
+    %p Start Date
+    %input.american_date_entry#start-datepicker{:type => "text", :name => "start_date", :value => @start_date}
+  %div.input-container
+    %p End Date
+    %input.american_date_entry#end-datepicker{:type => "text", :name => "end_date", :value => @end_date}
+  = submit_tag "Submit", :id => 'datepicker-submit-btn'
+
+= tabbed_panel do |tp|
+  = tp.add_tab('Table') do
+    %table.listing
+      %tr.week_row
+        %td
+        - @sundays.each do |sun|
+          %td
+            %h2= 'Week ending ' + sun
+      - @users.each do |u|
+        %tr.user_row
+          %td
+            %h2= u.name
+          - last = @sundays.length-1
```

Comment:
Ideally we like to keep calculations and database queries out of the view template as much as possible. Try to use the view template only to format HTML for data you have already calculated.    So, the queries and calculations being performed from line 37 to line 44 in this file would be better moved to the controller. (And if doing so makes the controller too big, we'd move the calculations to a Presenter object ... I can show you how to do that in person next week, it's much simpler than it sounds).

Since you need to compute this for each user, consider iterating the users back in the controller, and assembling a data structure that has everything the view needs.  For example, once constructed it might be a Ruby Hash object like:

    @users_with_hours = {
      evan => { :billable =>  <results of WorkUnitQuery for evan's billable hours>,
                :unbillable => <results of WorkUnitQuery for evan's unbillable hours>,
                :total => <results of WorkUnitQuery for evan's total hours>  },
      brooke => {
                 :billable => <similar>,
                 :unbillable => <similar>,
                 :total  => <similar> }           
      },
      ... other users ...
    } 

 ...And so on.

You'd also want to make sure the "check for missing weeks" capability had already been run before the view was rendered.

Then the view could simply iterate that data structure, something like
    - @users_with_hours.keys.each do |user|
       %tr
           %td= user.name
           - @sundays.each do |day|
                %td 
                   .billable=@users_with_hours[user][:billable][day]
                   .unbillable=  @users_with_hours[user][:unbillable][day]
                   .total =  @users_with_hours[user][:total][day]